scripts/dataset/export/index.js

```diff
@@ -0,0 +1,105 @@
+import fsApi from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+import archiver from 'archiver';
+
+import { instantiateVersionsStorageAdapter } from '../../../src/index.js';
+import * as renamer from '../../utils/renamer/index.js';
+import readme from '../assets/README.template.js';
+import logger from '../logger/index.js';
+
+const __dirname = path.dirname(fileURLToPath(import.meta.url));
+
+const fs = fsApi.promises;
+
+const ARCHIVE_FORMAT = 'zip'; // for supported formats, see https://www.archiverjs.com/docs/archive-formats
+
+export default async function generate({ archivePath, releaseDate }) {
+  const versionsStorageAdapter = await (instantiateVersionsStorageAdapter()).initialize();
+
+  const archive = await initializeArchive(archivePath);
+
+  await renamer.loadRules();
+
+  const services = new Set();
+  let firstVersionDate = new Date();
+  let lastVersionDate = new Date(0);
+
+  let index = 1;
+
+  for await (const version of versionsStorageAdapter.iterate()) {
+    const { content, fetchDate } = version;
+    const { serviceId, documentType } = renamer.applyRules(version.serviceId, version.documentType);
+
+    if (firstVersionDate > fetchDate) {
+      firstVersionDate = fetchDate;
+    }
+
+    if (fetchDate > lastVersionDate) {
+      lastVersionDate = fetchDate;
+    }
+
+    services.add(serviceId);
+
+    const versionPath = generateVersionPath({ serviceId, documentType, fetchDate });
+
+    logger.info({ message: versionPath, counter: index, hash: version.id });
```

Comment:
> Do we really need the counter? I am not sure I understand why we want to log every single entry. For `services-all`, for example, that's over 16 thousand lines. What do I gain from knowing that it is version number 13241 in the run?
>
> Removing the counter would simplify the code and remove the need for the `index` variable.

Reply:
I added this counter for two reasons:
- When manually running the export, it's handy to have a feedback with a progress log. As this can take some time (~15min for the full services-all versions history), it informs us that it is not blocked and shows us the progress.
- When it fails, it is very useful to find out where it fails so that we can understand why.